src/items.rs

```diff
@@ -1189,7 +1187,7 @@ pub(crate) fn format_trait(
         let bound_hi = bounds.last().unwrap().span().hi();
         let snippet = context.snippet(mk_sp(ident_hi, bound_hi));
         if contains_comment(snippet) {
-            return None;
+            return Err(RewriteError::Unknown);
```

Comment:
Not something we need to do in this PR, but it would be nice to add a new variant to `RewriteError` that communicates that rewriting failed because comments would be lost.

Reply:
I was wondering about this too. Or maybe something more general like KnownLimitation. But would such an error actually be useful over Unknown at runtime?

Reply:
For the potential comment loss, yes it would be. Eventually we'll want to communicate to end users that we failed to format their code because we weren't expecting to find a comment there, and in order to avoid removing the comment we're leaving that AST node unformatted.

Reply:
In some cases we already communicate that to end users, but it's not always consistent.

Reply:
Makes sense!